Automattic/wp-calypso
# Add default jetpack icon on iconless sites in My Purchases

## Proposed Changes

* On My Purchases, check if site has an icon. If it is a Jetpack Purchase and the site has no icon, default to the Jetpack Icon

## Why are these changes being made?

Jetpack purchases are indistinguishable if the user has Wordpress plans and Jetpack plans on their account. 

More info: p1701265535943769/1701262414.362659-slack-C01264051NE

## Testing Instructions

1. If you don't already have one, create a new Jurassic Ninja site, go to `/wp-admin/options-general.php`, and set the site icon to anything you want. I used this 
![image](https://github.com/user-attachments/assets/05847e13-65e3-4e8e-a980-3042454c85ea)
![image](https://github.com/user-attachments/assets/0c5343a4-c691-44c7-8861-c874db1ec162)
2. Purchase any Jetpack plan or product for this site
3. Now create another Jurassic Ninja site and don't add a site icon
4. Purchase any Jetpack plan or product for this site as well
5. Now purchase another Jetpack plan or product in a siteless checkout flow and do not assign the license to a site
6. Open the calypso blue live link or checkout this branch locally and go to `/me/purchases`
7. You should see a purchase with the icon you chose for the site, an error icon for the disconnected license, and the Jetpack Icon for the purchase made on your site without an icon
![image](https://github.com/user-attachments/assets/43f918ff-9a37-4fc2-984b-14011db89604)


## Pre-merge Checklist
